kbfsgit/runner.go

```diff
@@ -37,6 +40,26 @@ const (
 	publicName  = "public"
 	privateName = "private"
 	teamName    = "team"
+
+	// localRepoRemoteName is the name of the remote that gets added
+	// locally to the config of the KBFS bare repo, pointing to the
+	// git repo stored at the `gitDir` passed to `newRunner`.
+	//
+	// In go-git, there is no way to hook two go-git.Repository
+	// instances together to do fetches/pulls between them. One of the
+	// two repos has to be defined as a "remote" to the other one in
+	// order to use the nice Fetch and Pull commands. (There might be
+	// other more involved ways to transfer objects manually
+	// one-by-one, but that seems like it would be pretty sad.)
+	//
+	// Since there is no standard remote protocol for keybase yet
+	// (that's what we're building!), it's not supported by go-git
+	// itself. That means our only option is to treat the local
+	// on-disk repo as a "remote" with respect to the bare KBFS repo,
+	// and do everything in reverse: for example, when a user does a
+	// push, we actually fetch from the local repo and write the
+	// objects into the bare repo.
+	localRepoRemoteName = "local"
```

Comment:
I just realized a thing: in a normal git setup, we'd have have `origin/master` ref stored in local repo (which allows you do things like `git reset --hard origin/master`). In this reverse setup, would go-git store refs like `local/master` in the bare repo, which points to the local repo on disk in the bare repo? If so, do we need to clean them up after each push/fetch?

Reply:
No, from my tests it doesn't seem like go-git does that when you simply call `Fetch`.  I think you have to explicitly add a reference to get it to save it to the repo.  (Or maybe that's a bare repo thing?)

In any case, in this PR we're not even calling `Fetch` on the remote, we're calling `Push`, so it doesn't modify the bare repo at all (except for adding the remote).